### PR TITLE
New Method TH1::SetColors

### PR DESCRIPTION
**Pull request by couet**
Implement the method requested here: https://root-forum.cern.ch/t/a-few-new-member-functions-for-various-classes/59434/3

By default: linecolor = markercolor = fillcolor = -1
If a color is < 0 this method does not change the corresponding color if positive or null it set the color.

For instance:

```
 h->SetColor(kRed, kRed);
```
will set the line color and the marker color to red.

### CODE REVIEW

File: hist/hist/src/TH1.cxx [hunk 1]
```diff
@@ -4454,6 +4454,26 @@ Double_t TH1::GetEffectiveEntries() const
    return (s[1] ? s[0]*s[0]/s[1] : TMath::Abs(s[0]) );
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Shortcut to set the four histogram colors with a single call.
```
Comment (linev) [inline hist/hist/src/TH1.cxx:4458]: Method changes **three** colors - not four

File: hist/hist/src/TH1.cxx [hunk 2]
```diff
@@ -4454,6 +4454,26 @@ Double_t TH1::GetEffectiveEntries() const
...
+/// By default: linecolor = markercolor = fillcolor = -1
+/// If a color is < 0 this method does not change the corresponding color if positive or null it set the color.
+///
+/// For instance:
+/// ~~~ {.cpp}
+/// h->SetColor(kRed, kRed);
+/// ~~~
+/// will set the line color and the marker color to red.
+
+void TH1::SetColor(Color_t linecolor, Color_t markercolor, Color_t fillcolor)
```
Comment (linev) [inline hist/hist/src/TH1.cxx:4469]: May be one should call it `SetColors` or `SetHistColors`?

**Review by linev — CHANGES_REQUESTED** [review]
Beside cosmetics in comment - one better use plural form in method name

File: hist/hist/src/TH1.cxx (hunk 1 above)
Comment (couet) [reply hist/hist/src/TH1.cxx:4458]: good point

File: hist/hist/src/TH1.cxx (hunk 2 above)
Comment (couet) [reply hist/hist/src/TH1.cxx:4469]: As it is a TH1 method it is quite obvious it is a histogram method so there is no need to add "Hist" in the name (I checked that none of the TH1 methods have "Hist" in their name). However, adding an "s" is a good idea.

File: hist/hist/src/TH1.cxx [hunk 3]
```diff
@@ -4454,6 +4454,26 @@ Double_t TH1::GetEffectiveEntries() const
...
 
+////////////////////////////////////////////////////////////////////////////////
+/// Shortcut to set the three histogram colors with a single call.
+///
+/// By default: linecolor = markercolor = fillcolor = -1
+/// If a color is < 0 this method does not change the corresponding color if positive or null it set the color.
+///
+/// For instance:
+/// ~~~ {.cpp}
+/// h->SetColor(kRed, kRed);
```
Comment (linev) [inline hist/hist/src/TH1.cxx:4465]: Also here `SetColors`

Comment (couet) [reply hist/hist/src/TH1.cxx:4465]: sure ! .. stupid me

**Review by linev — APPROVED** [review]

**Pull request merged**